Lib/subprocess.py

```diff
@@ -1113,8 +1114,10 @@ def _get_handles(self, stdin, stdout, stderr):
                 p2cread = _winapi.GetStdHandle(_winapi.STD_INPUT_HANDLE)
                 if p2cread is None:
                     p2cread, _ = _winapi.CreatePipe(None, 0)
-                    p2cread = Handle(p2cread)
-                    _winapi.CloseHandle(_)
+                    try:
+                        p2cread = Handle(p2cread)
+                    finally:
+                        _winapi.CloseHandle(_)
```

Comment:
I'm not sure that `try/finally` is worth it here. `Handle()` may fail only due to memory shortage, and if it does, `p2cread` is still leaked, even though `_` will be closed. Moreover, three lines below potential failures of `Handle()` are not handled (sorry for the pun).

Reply:
Agree. These two changes were meant as a little "extra". I'll remove them for the sake of "pureness" and perhaps address that in another wider PR.

Reply:
The try/finally is cheap if no exception is thrown, but you're right that it would be more correct to also have an `except:` close that closed `p2cread`.

The `PIPE` cases are more complicated, so I'd be OK with just added a comment on them noting that they can leak handles if `Handle` throws an exception.